### PR TITLE
EVA-2470 Extract and ingest remapping improvements

### PR DESCRIPTION
**Pull request by andresfsilva**
- [x] Filter out variants with the mapWeight flag during the extraction
- [x] Include createdDate (date of ssid creation) in extraction and ingestion
- [x] RS id is not required to ingest remapping results
- [x] Change taxonomy key to be the same in the remapping extract and ingest

Remapping metadata will be implemented in a separate PR

### CODE REVIEW

File: eva-accession-core/src/main/java/uk/ac/ebi/eva/accession/core/model/SubmittedVariant.java [hunk 1]
```diff
@@ -232,71 +243,28 @@ public void setValidated(Boolean validated) {
...
+                Objects.equals(referenceSequenceAccession, that.referenceSequenceAccession) &&
+                Objects.equals(projectAccession, that.projectAccession) &&
+                Objects.equals(contig, that.contig) &&
+                Objects.equals(referenceAllele, that.referenceAllele) &&
+                Objects.equals(alternateAllele, that.alternateAllele) &&
+                Objects.equals(clusteredVariantAccession, that.clusteredVariantAccession) &&
+                Objects.equals(supportedByEvidence, that.supportedByEvidence) &&
+                Objects.equals(assemblyMatch, that.assemblyMatch) &&
+                Objects.equals(allelesMatch, that.allelesMatch) &&
+                Objects.equals(validated, that.validated);
```
Comment (sundarvenkata-EBI) [inline eva-accession-core/src/main/java/uk/ac/ebi/eva/accession/core/model/SubmittedVariant.java:260]: Have you verified the equivalence of these statements against the previous (more verbose) code? I am hoping the complex ternary conditionals for attributes like supportedByEvidence in the previous code are covered in this terse validation..

File: eva-accession-core/src/main/java/uk/ac/ebi/eva/accession/core/model/SubmittedVariant.java [hunk 2]
```diff
@@ -232,71 +243,28 @@ public void setValidated(Boolean validated) {
...
-        result = 31 * result + alternateAllele.hashCode();
-        result = 31 * result + (clusteredVariantAccession != null ? clusteredVariantAccession.hashCode() : 0);
-        result = 31 * result + (supportedByEvidence != null ? supportedByEvidence.hashCode() : 0);
-        result = 31 * result + (assemblyMatch != null ? assemblyMatch.hashCode() : 0);
-        result = 31 * result + (allelesMatch != null ? allelesMatch.hashCode() : 0);
-        result = 31 * result + (validated != null ? validated.hashCode() : 0);
-        return result;
+        return Objects.hash(referenceSequenceAccession, taxonomyAccession, projectAccession, contig, start,
+                            referenceAllele, alternateAllele, clusteredVariantAccession, supportedByEvidence,
+                            assemblyMatch, allelesMatch, validated);
```
Comment (sundarvenkata-EBI) [inline eva-accession-core/src/main/java/uk/ac/ebi/eva/accession/core/model/SubmittedVariant.java:267]: Similar to previous comment, have you verified the equivalence of this hash against the one obtained with more verbose code?

File: eva-accession-core/src/main/java/uk/ac/ebi/eva/accession/core/model/eva/SubmittedVariantEntity.java [hunk 3]
```diff
@@ -107,13 +109,39 @@ public SubmittedVariantEntity(Long accession, String hashedMessage, String refer
...
+        this.contig = contig;
+        this.start = start;
+        this.referenceAllele = referenceAllele;
+        this.alternateAllele = alternateAllele;
+        this.clusteredVariantAccession = clusteredVariantAccession;
+
+        setSupportedByEvidence(supportedByEvidence);
+        setAssemblyMatch(assemblyMatch);
+        setAllelesMatch(allelesMatch);
+        setValidated(validated);
```
Comment (sundarvenkata-EBI) [inline eva-accession-core/src/main/java/uk/ac/ebi/eva/accession/core/model/eva/SubmittedVariantEntity.java:133]: Could we invoke the previous constructor in lieu of all the above statements?

File: eva-remapping-get-source/src/test/java/uk/ac/ebi/eva/remapping/source/batch/processors/VariantToVariantContextProcessorTest.java [hunk 4]
```diff
@@ -65,6 +66,7 @@ private SubmittedVariantEntity buildVariant(String chr1, int start, String refer
         SubmittedVariantEntity variant = new SubmittedVariantEntity(SS_ID, "hash", "assembly", taxonomy, project, chr1,
                                                                     start, reference, alternate, RS_ID,
                                                                     false, false, false, false, 1);
+        variant.setCreatedDate(LocalDateTime.now());
```
Comment (sundarvenkata-EBI) [inline eva-remapping-get-source/src/test/java/uk/ac/ebi/eva/remapping/source/batch/processors/VariantToVariantContextProcessorTest.java:69]: Are there downsides to using a single timestamp per run for easier debugging?

**Review by sundarvenkata-EBI — APPROVED** [review]

**Review by tcezard — APPROVED** [review]

File: eva-accession-core/src/main/java/uk/ac/ebi/eva/accession/core/model/SubmittedVariant.java (hunk 1 above)
Comment (andresfsilva) [reply eva-accession-core/src/main/java/uk/ac/ebi/eva/accession/core/model/SubmittedVariant.java:260]: Yes, they are equivalent

File: eva-accession-core/src/main/java/uk/ac/ebi/eva/accession/core/model/eva/SubmittedVariantEntity.java (hunk 3 above)
Comment (andresfsilva) [reply eva-accession-core/src/main/java/uk/ac/ebi/eva/accession/core/model/eva/SubmittedVariantEntity.java]: Done

File: eva-remapping-get-source/src/test/java/uk/ac/ebi/eva/remapping/source/batch/processors/VariantToVariantContextProcessorTest.java (hunk 4 above)
Comment (andresfsilva) [reply eva-remapping-get-source/src/test/java/uk/ac/ebi/eva/remapping/source/batch/processors/VariantToVariantContextProcessorTest.java]: Done

**Review by sundarvenkata-EBI — APPROVED** [review]

**Pull request merged**